From d5bccab2a85f2b9f4346b72c6e4721452168a9ab Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 19 Apr 2020 23:09:22 +0300
Subject: [PATCH] [SW-0] preboot: version: use register from regmap

The mmPSOC_GLOBAL_CONF_SCRATCHPAD_28 is used for providing the Preboot
version offset for the host driver.
Use the common name provided by regmap for this register.
This way, if some device will use a differen register for the same
purpose, it will be transparent for version code.

Change-Id: I6182c6752233a009e8360f507970fc0417a44bed
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/preboot.c b/app/preboot.c
index 46aa25e..e00037a 100644
--- a/app/preboot.c
+++ b/app/preboot.c
@@ -130,7 +130,7 @@
 __attribute__((weak)) void preboot_version(void)
 {
 	unsigned long offset = (unsigned long) preboot_btl_ver_str;
-	u32 btlver = regmap_readl(mmPSOC_GLOBAL_CONF_SCRATCHPAD_28);
+	u32 btlver = regmap_readl(mmPREBOOT_VER_OFFSET);
 	int i, btl_ver_sz;
 
 	preboot_puts("\n");
@@ -158,8 +158,7 @@
 			break;
 	}
 
-	regmap_writel((u32)(offset & ~SDRAM_BASE),
-			mmPSOC_GLOBAL_CONF_SCRATCHPAD_28);
+	regmap_writel((u32)(offset & ~SDRAM_BASE), mmPREBOOT_VER_OFFSET);
 }
 #else	/* TARGET_IS_TESTER */
 __attribute__((weak)) void preboot_version(void) {}
